### PR TITLE
fix(config/jobs/build-drivers): set correct diff filename regex

### LINKED ISSUE #1072: Build-drivers job not correctly triggered
**Describe the bug**

After latest kernel-crawler run (https://github.com/falcosecurity/kernel-crawler/pull/112), and the PR to update dbg configs (https://github.com/falcosecurity/test-infra/pull/1068), `build-new-drivers` jobs for x86_64 ubuntu-* were not triggered.

This is the first time the issue happened and the jobs were untouched since last week (when they ran fine).
Moreover, new configs for ubuntu were actually created, as looked up through:
```
git diff --name-only --diff-filter=A b457718870a62131a93689d83bfaede4f0c02554 driverkit/config/4.0.0+driver/x86_64/  | grep ubuntu
```
Whose output is:
```
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws-5.15_5.15.0-1033-aws-5.15_37~20.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws-5.19_5.19.0-1022-aws-5.19_23~22.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws-5.4_5.4.0-1099-aws-5.4_107~18.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws_4.15.0-1153-aws_166.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws_5.15.0-1033-aws_37.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws_5.19.0-1022-aws_23.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws_5.4.0-1099-aws_107.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-aws_6.2.0-1002-aws_2.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-azure_6.2.0-1002-azure_2.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-gcp-5.4_5.4.0-1102-gcp-5.4_111~18.04.2.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-gcp_5.19.0-1019-gcp_21.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-gcp_6.2.0-1004-gcp_4.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-generic_4.15.0-208-generic_220.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-generic_5.15.0-69-generic_76.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-generic_5.19.0-38-generic_39.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-generic_5.4.0-146-generic_163.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-generic_6.2.0-18-generic_18.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-gke_5.15.0-1030-gke_35.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-gke_5.4.0-1096-gke_103.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-gkeop_5.4.0-1066-gkeop_70.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-hwe-5.15_5.15.0-69-hwe-5.15_76~20.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-hwe-5.19_5.19.0-38-hwe-5.19_39~22.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-hwe-5.4_5.4.0-146-hwe-5.4_163~18.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-intel-iotg-5.15_5.15.0-1027-intel-iotg-5.15_32~20.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-kvm_6.2.0-1002-kvm_2.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-lowlatency-hwe-5.15_5.15.0-69-lowlatency-hwe-5.15_76~20.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-lowlatency-hwe-5.19_5.19.0-1021-lowlatency-hwe-5.19_22~22.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-lowlatency_5.15.0-69-lowlatency_76.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-lowlatency_5.19.0-1021-lowlatency_22.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-lowlatency_6.2.0-1002-lowlatency_2.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-oracle-5.15_5.15.0-1032-oracle-5.15_38~20.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-oracle-5.4_5.4.0-1098-oracle-5.4_107~18.04.1.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-oracle_5.15.0-1032-oracle_38.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-oracle_5.4.0-1098-oracle_107.yaml
driverkit/config/4.0.0+driver/x86_64/ubuntu-oracle_6.2.0-1002-oracle_2.yaml
```

Therefore, we'd expect:
* ubuntu-aws
* ubuntu-azure
* ubuntu-gcp
* ubuntu-generic
* ubunut-gke

to be triggered.

### PR DESCRIPTION
**Pull request by maxgio92**
The star character makes the whole expression to matche even when the preceding match do not.

Fixes #1072

### CODE REVIEW

**Comment by maxgio92** [comment]
PTAL @FedeDP

**Comment by maxgio92** [comment]
Note: there is an assumption on the DBG filename. The filenames always have an extension.

File: config/jobs/build-drivers/build-new-bottlerocket.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ presubmits:
     agent: kubernetes
     branches:
       - ^master$
-    run_if_changed: '^driverkit/config/[a-z0-9.+-]{5,}/(.+/)?bottlerocket*'
+    run_if_changed: '^driverkit/config/[a-z0-9.+-]{5,}/(.+/)?bottlerocket.+'
```
Comment (FedeDP) [inline config/jobs/build-drivers/build-new-bottlerocket.yaml:9]: Also, can we use `_.*` everywhere? We always expect an `_` after the target name.

Comment (FedeDP) [reply config/jobs/build-drivers/build-new-bottlerocket.yaml]: Done myself :)

Comment (maxgio92) [reply config/jobs/build-drivers/build-new-bottlerocket.yaml]: Thank you @FedeDP

**Pull request merged**